cirq/google/api/v2/program.proto

```diff
@@ -64,6 +66,11 @@ message Moment {
   // All of the operations in the moment. Each operation must act on different
   // qubits.
   repeated Operation operations = 1;
+
+  // As 'operations', but may include CircuitOperations (i.e. operations that
+  // contain subcircuits). If 'operations' is non-empty, this field will be
+  // ignored.
+  repeated GenericOperation generic_operations = 2;
```

Comment:
What is the purpose of adding GenericOperation?  Why not just put repeated CircuitOperations straight into Moment?  I am not sure GenericOperation really adds much here and the name is not very descriptive either.

Reply:
Using repeated CircuitOperation would either (1) imply that a moment can only contain either Operations or CircuitOperations, or (2) require some implicit ordering between Operations and CircuitOperations. Both feel like improper uses of Moment.

What I would've preferred is to call this `Operation`, with the current operation renamed to `GateOperation`...but past experience with proto-renaming tells me that would be a nightmare for backwards-compatibility.

Reply:
> Is the idea here that the `operations` field should be deprecated and we should put everything into `generic_operations` instead {...}?

That was my original thought, yes.

> I think just have two lists of `operations` and `circuit_operations` right here in Moment might make more sense {...}

It didn't occur to me in my previous comment, but I guess this would be possible since there isn't any ordering of operations inside a Moment. I'll go ahead with this change.

Reply:
Done.